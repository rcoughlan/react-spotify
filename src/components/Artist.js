import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
    Card,
    Col
} from 'react-bootstrap';

class Artist extends Component {

    render() {
        let genres = this.props.genres.map( (genre, index) => 
            <Card.Text key={index} className="genres" alt={'genre ' + genre}>{genre}</Card.Text>
        )

        return (

            <Col>
                <a href={this.props.link} target="_blank" className="spotify-link">
                    <Card className="music-card">
                    <Card.Img className="thumbnail" alt={this.props.name + ' artist image'} variant="top" src={this.props.image} />
                        <Card.Body>
                        <Card.Title className="artist-name" alt={'artist: ' + this.props.name}>{this.props.name}</Card.Title>
                        {genres}
                        </Card.Body>
                    </Card>
                </a>
            </Col>

        );
    }
}

Artist.propTypes = {
    name: PropTypes.string,
    image: PropTypes.string,
    index: PropTypes.number,
    genres: PropTypes.arrayOf(PropTypes.string),
    link: PropTypes.string
}

export default Artist;