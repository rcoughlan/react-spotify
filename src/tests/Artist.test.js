import Artist from '../components/Artist';
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

describe('Artist Component', () => {

    let artistObject = {
        name: 'Arctic Monkeys',
        genres: ['rock', 'indie', 'sheffield indie'],
        image: 'https://i.scdn.co/image/ed0552e9746ed2bbf04ae4bcb5525700ca31522d',
        link: 'https://open.spotify.com/artist/7Ln80lUS6He07XvHI8qqHH'
    }

    it('matches the snapshot', () => {
        const tree = renderer.create(<Artist name={artistObject.name} genres={artistObject.genres} image={artistObject.image} link={artistObject.link}/>).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render a component with a thumbnail image with correct alt tag', () => {
        const wrapper = shallow(<Artist name={artistObject.name} genres={artistObject.genres} image={artistObject.image} link={artistObject.link}/>);
        const thumbnail = wrapper.find('.thumbnail');
        expect(thumbnail).toHaveLength(1);
        
        expect(thumbnail.prop('alt')).toBe(artistObject.name + ' artist image');
        expect(thumbnail.prop('src')).toBe(artistObject.image);
    });
    
    it('card should link to the artist on spotify', () => {
        const wrapper = shallow(<Artist name={artistObject.name} genres={artistObject.genres} image={artistObject.image} link={artistObject.link}/>);
        const thumbnail = wrapper.find('.spotify-link');
        expect(thumbnail).toHaveLength(1);
        
        expect(thumbnail.prop('href')).toBe(artistObject.link);
        expect(thumbnail.prop('target')).toBe('_blank');
    });
    
    it('should have an artist title', () => {
        const wrapper = shallow(<Artist name={artistObject.name} genres={artistObject.genres} image={artistObject.image} link={artistObject.link}/>);
        const artist = wrapper.find('.artist-name');
        expect(artist).toHaveLength(1);
        expect(artist.prop('alt')).toBe('artist: ' + artistObject.name);
    });
    
    // it('should have an list of genres', () => {
    //     const wrapper = shallow(<Artist name={artistObject.name} genres={artistObject.genres} image={artistObject.image} link={artistObject.link}/>);
    //     const genres = wrapper.find('p.genres');
    //     expect(genres.prop('alt')).toBe('genres: ' + artistObject.genres);
    // });
});