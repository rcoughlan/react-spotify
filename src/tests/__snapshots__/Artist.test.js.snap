// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artist Component matches the snapshot 1`] = `
<div
  className="col"
>
  <a
    className="spotify-link"
    href="https://open.spotify.com/artist/7Ln80lUS6He07XvHI8qqHH"
    target="_blank"
  >
    <div
      className="music-card card"
    >
      <img
        alt="Arctic Monkeys artist image"
        className="card-img-top thumbnail"
        src="https://i.scdn.co/image/ed0552e9746ed2bbf04ae4bcb5525700ca31522d"
      />
      <div
        className="card-body"
      >
        <div
          alt="artist: Arctic Monkeys"
          className="artist-name card-title h5"
        >
          Arctic Monkeys
        </div>
        <p
          alt="genre rock"
          className="genres card-text"
        >
          rock
        </p>
        <p
          alt="genre indie"
          className="genres card-text"
        >
          indie
        </p>
        <p
          alt="genre sheffield indie"
          className="genres card-text"
        >
          sheffield indie
        </p>
      </div>
    </div>
  </a>
</div>
`;
